danceradvent/public/articles/2015/3-medium-scale-part-1.pod

```diff
@@ -0,0 +1,100 @@
+=encoding utf8
+
+=head1 Medium-Scale Dancer, Part 1: Modules
+
+=head2 Or: How to Get Beyond Small-Scale in Dancer Application Design
+
+When you generate a Dancer app with C<dancer2 -a App>, you get a very
+nicely structured and useful starting point. As long as your app remains
+fairly small, you can work with this structure as-is clear through to
+completion. In that sense, the generated app is production-ready.
+
+The problem comes when you start getting to thousands of lines of Perl
+code, dozens of view files, and hundreds of routes, stretching the stock
+app design beyond its natural limits. This series of articles gives you
+one way to extend beyond this base.
```

Comment:
I backed off on the "hundreds of routes" bit. On re-reading it, it's bothered me a bit because it implies that the number of routes might reasonably exceed the number of template files by an order of magnitude.

I don't find that to be the case; in fact, the ratio of routes to template files happens to be within a few percent of 50/50 in my Dancer application. I find that the pressures pushing for imbalance in one direction are met by opposing pressures: for every route that pulls in multiple templates to build the final page, there are multiple routes that don't use template files at all.